From 934f51d166848c275828f14ad5cc29fe1c3b578a Mon Sep 17 00:00:00 2001
From: Paul Smith <psmith@gnu.org>
Date: Mon, 14 Nov 2011 00:18:38 +0000
Subject: In very obscure situations we may write the free token back to the pipe.

Don't do that.  I couldn't come up with a repro case for this!
---
(limited to 'main.c')

diff --git a/main.c b/main.c
index 75eb494..d260859 100644
--- a/main.c
+++ b/main.c
@@ -3116,7 +3116,8 @@ clean_jobserver (int status)
                "INTERNAL: Exiting with %u jobserver tokens (should be 0)!",
                jobserver_tokens);
       else
-        while (jobserver_tokens--)
+        /* Don't write back the "free" token */
+        while (--jobserver_tokens)
           {
             int r;
 
--
cgit v0.9.0.2
